src/views/SendToken/SendToken.tsx

```diff
@@ -289,37 +299,47 @@ export function SendToken({ identity, onSuccess }: Props): JSX.Element {
     })();
   }, [amountBN, identity, recipient, tipBN]);
 
-  const handleAmountInput = useCallback((event) => {
-    const { value } = event.target;
+  const handleAmountInput = useCallback((event: FormEvent) => {
+    const { value } = event.target as HTMLInputElement;
     if (value.match(nonNumberCharacters)) {
-      event.target.value = value.replace(nonNumberCharacters, '');
+      (event.target as HTMLInputElement).value = value.replace(
+        nonNumberCharacters,
+        '',
+      );
     }
     setAmount(value);
   }, []);
 
-  const handleAmountBlur = useCallback((event) => {
-    const { value } = event.target;
-    if (value.length === 0) {
-      setAmount(value);
-      return;
-    }
-    const parsedValue = parseFloatLocale(value);
-    if (Number.isNaN(parsedValue)) {
-      return;
-    }
-    const femtoKilts = numberToBN(parsedValue);
-    const formatted = formatKiltInput(femtoKilts);
-    event.target.value = formatted;
-    setAmount(formatted);
-  }, []);
+  const handleAmountBlur = useCallback(
+    (event: FocusEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+      if (value.length === 0) {
+        setAmount(value);
+        return;
+      }
+      const parsedValue = parseFloatLocale(value);
+      if (Number.isNaN(parsedValue)) {
+        return;
+      }
+      const femtoKilts = numberToBN(parsedValue);
+      const formatted = formatKiltInput(femtoKilts);
+      event.target.value = formatted;
+      setAmount(formatted);
+    },
+    [],
+  );
 
   const handleAllInClick = useCallback(
-    (event) => {
+    (event: MouseEvent) => {
```

Comment:
Would we need the type casting below if we specify the type?
```suggestion
    (event: MouseEvent<HTMLButtonElement>) => {
```

Reply:
In this case yes, without the type casting I get this error:

`Property 'form' does not exist on type 'EventTarget'.`

Reply:
Here we can also use `currentTarget`, but will have to check that form is not null.